### PR TITLE
Reproduce arXiv:2207.05612 Figure 2(a)

### PR DESCRIPTION
**Pull request by refraction-ray**
Implemented reproduction of Figure 2(a) from arXiv:2207.05612 using TensorCircuit's MPS simulator.
- Created `examples/reproduce_papers/2022_dmrg_circuit_simulation/`.
- Implemented `main.py` to generate RQC and simulate with varying bond dimensions.
- Verified results against exact simulation.
- Verified code with `black` and `pylint`.

---
*PR created automatically by Jules for task [4764008025812483835](https://jules.google.com/task/4764008025812483835) started by @refraction-ray*

### CODE REVIEW

File: examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+import tensorcircuit as tc
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Use numpy backend for broad compatibility
+K = tc.set_backend("numpy")
```
Comment (refraction-ray) [inline examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py]: make sure the implementation work for jax backend

File: examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+Link: https://arxiv.org/abs/2207.05612
+
+Method:
+- Single-site DMRG with JAX backend.
+- Qubit Grouping: Columns of the grid are mapped to single MPS sites.
+- Vertical Contraction: Environment tensors L and R carry "gate legs".
+- Analytic Update: New site tensor is computed by contracting L, R, and the circuit column.
+"""
+
+import logging
```
Comment (refraction-ray) [inline examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py]: 致命漏洞 1：缺失正交化约束 (Missing Orthogonalization)这是当前代码最严重的问题。论文在提出公式 (20) 前，有一句极其重要的话（Eq. 19 上方）："Before doing the optimization, we need to enforce the fact that the MPS is a normalized state... This is best done by performing a series of QR factorizations... to bring the MPS in the so-called 'orthogonal form'."为什么必须做 QR 分解？你使用的公式 E = E / norm 是基于一个严格的数学前提推导出来的：除当前正在优化的张量外，MPS 的其余所有张量必须构成正交基（即满足 $\langle \Psi | \Psi \rangle = \text{Tr}(M^{(\tau)} M^{(\tau)*})$）。在你的 run_dmrg 中，你直接把算出的 E 赋值给了 mps_tensors[i]，然后直接移动到 i+1。这破坏了 MPS 的规范形式（Canonical form），导致你在算下一步的 L 时，基底不再正交，目标函数的二次型约束被打破，优化的结果就变成了无意义的噪声。如何修复？在 Sweep 过程中，更新完当前节点后，必须用 QR 分解（左到右）或 LQ 分解（右到左）将多余的自由度（即 $R$ 矩阵）推给下一个节点。Python# 左到右 Sweep 时的修正逻辑示例
for i in range(self.cols):
    L = self.get_L(i)
    R = self.get_R(i)
    E, _ = self.optimize_site(i, L, R)
    
    if i < self.cols - 1:
        # 1. 铺平张量以进行 QR 分解: (左键维 * 物理维, 右键维)
        E_mat = jnp.reshape(E, (-1, E.shape[-1]))
        
        # 2. QR 分解
        Q, R_mat = jnp.linalg.qr(E_mat)
        
        # 3. 当前节点变成正交的 Q
        self.mps_tensors[i] = jnp.reshape(Q, (E.shape[0], E.shape[1], -1))
        
        # 4. 把 R_mat 吸收到下一个节点 (重要！)
        next_A = self.mps_tensors[i+1]
        self.mps_tensors[i+1] = jnp.tensordot(R_mat, next_A, axes=[[1], [0]])
    else:
        # 最后一个节点不需要 QR，直接保留范数
        self.mps_tensors[i] = E
(注：从右到左的 Sweep 需要做对称的 LQ 分解，把 L 矩阵乘给左边的 i-1 节点。)严重架构偏离 2：一次性全局压缩 vs. 逐块（$K$ 层）压缩仔细观察你的 update_L 和 update_R 逻辑，你会发现你的代码实际上在做这样一件事：将一整个深度为 $D$ 的完整线路直接作用在初始态 $|00\dots0\rangle$ 上，然后试图一次性用 MPS 拟合出最终态。你的做法：在垂直收缩时，循环 for k in range(self.num_layers) 一口气吞下了所有层。如果线路深度 $D=20$，那么你在计算环境 $L$ 时，网络最宽处的物理指标将高达 $2^{\text{rows} \times 20}$。这正是“薛定谔”模拟的指数爆炸！深度稍微大一点，显存立刻 OOM（内存溢出）。论文的做法（见图 4A 和 4C）：算法的核心理念是步进压缩（Compression Step）。假设当前有了一个近似到第 $D$ 层的旧 MPS：$|\Psi_{old}\rangle$。仅拿出接下来的 $K$ 层量子门（例如 $K=2$）。用 DMRG 寻找一个新的 MPS $|\Psi_{new}\rangle$，使其最大程度逼近 $U_{K} \dots U_1 |\Psi_{old}\rangle$。这里的环境 $F^{(\tau)}$ 的垂直收缩，是夹在 $|\Psi_{new}\rangle$、$K$ 层量子门、和 $|\Psi_{old}\rangle$ 之间的。它的收缩复杂度仅为 $\mathcal{O}(\chi^2 2^{\text{rows} \times K})$，与总深度 $D$ 彻底无关。如何修复？你的 EnvManager 需要改造为接收一个 old_mps_tensors（即上一轮压缩的结果），而不是每次都从底部（Final Contraction with |0>）开始算。每次调用 run_dmrg 时，传入的 layers 只应包含极少的层数（比如 $K=2$）。整个模拟的最外层应该是一个遍历线路所有深度的循环：Python# 伪代码：对齐论文的 K=2 步进架构
mps = init_state_0() # 初始 |00...0> MPS
K = 2
for depth_step in range(0, total_depth, K):
    k_layers = circuit_layers[depth_step : depth_step+K]
    
    # 用当前的 mps 作为 old_mps，加上 k_layers 进行拟合
    env_manager = EnvManager(old_mps=mps, layers=k_layers, ...)
    
    # 扫 2 到 4 次
    fids = env_manager.run_dmrg(sweeps=4)
    
    # 新算出的 tensors 作为下一轮的输入
    mps = env_manager.mps_tensors 

另外，如果有生成的图片，请全部放在 outputs 文件夹，并在 meta.yaml 注明

File: examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py [hunk 3]
```diff
@@ -0,0 +1,619 @@
+"""
+Reproduction of "A density-matrix renormalization group algorithm for simulating
```
[the remark this answers is not in the document]
Comment (refraction-ray) [reply examples/reproduce_papers/2022_dmrg_circuit_simulation/main.py]: the figure is still not cleaned, one in examples/ and there is a result.png in outputs? what are them, be mindful, clean up everything, and black, pytest, too
